src/main/java/com/minecolonies/colony/Colony.java

```diff
@@ -820,7 +880,31 @@ private void addBuilding(AbstractBuilding building)
     }
 
     /**
-     * Creates a building from a tile entity and adds it to the colony.
+     * Add a Building to the Colony.
+     *
+     * @param field      Field to add to the colony.
+     */
+    private void addField(Field field)
+    {
+        fields.put(field.getID(), field);
+    }
+
+    /**
+     * Creates a field from a tile entity and adds it to the colony.
+     * @param inventoryField the inventory of the field.
+     * @param inventoryPlayer the inventory of the player.
+     * @param pos Position where the field has been placed.
+     * @param world the world of the field.
+     */
+    public void addNewField(InventoryField inventoryField, InventoryPlayer inventoryPlayer, BlockPos pos, World world)
+    {
+        Field field = new Field(inventoryField, inventoryPlayer, world, pos);
+        addField(field);
+        ColonyManager.markDirty();
+    }
+
+    /**
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) First sentence should end with a period. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=checkstyle%3AJavadoc_Style)